### PR TITLE
Bloom Pass Inspired from Unreal

### PR DESCRIPTION
**Pull request by spidersharma03**
This Pass consists of the following passes inside::
- Extract Bright Pass
- Mip Map blur Pass
- Combine Pass

### CODE REVIEW

**Comment by WestLangley** [comment]
Can you target the dev branch, please?

**Comment by bhouston** [comment]
I also notice that this example doesn't handle resize properly, one needs to add a Pass.setSize function like this one:

https://github.com/mrdoob/three.js/blob/master/examples/js/postprocessing/ManualMSAARenderPass.js#L65

**Comment by bhouston** [comment]
THis is a really nice bloom pass.  But yeah the PR needs to target the dev branch.

File: examples/webgl_postprocessing_unreal_bloom.html [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+
+			function onWindowResize() {
+
+				var width = window.innerWidth;
+				var height = window.innerHeight;
+
+				camera.aspect = width / height;
+				camera.updateProjectionMatrix();
+
+				renderer.setSize( width, height );
```
Comment (bhouston) [inline examples/webgl_postprocessing_unreal_bloom.html]: add composer.setSize( width, height );

File: examples/webgl_postprocessing_unreal_bloom.html [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+				controls.target.set( 0, 0, 0 );
+				controls.update();
+
+				window.addEventListener( 'resize', onWindowResize, false );
+
+				var gui = new dat.GUI();
+
+				gui.add( params, 'exposure', 0.1, 2 );
+				gui.add( params, 'bloomThreshold', 0.0, 1.0 ).onChange( function(value) {
+		        bloomPass.threshold = Number(value);
```
Comment (bhouston) [inline examples/webgl_postprocessing_unreal_bloom.html]: can just move this into the render() call.  Less code.  Same with bloomPass.strength/radius.

File: examples/webgl_postprocessing_unreal_bloom.html [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+
+				renderer.setPixelRatio( window.devicePixelRatio );
+				renderer.setSize( window.innerWidth, window.innerHeight );
+				renderer.shadowMap.enabled = true;
+				container.appendChild( renderer.domElement );
+
+				renderScene = new THREE.RenderPass(scene, camera);
+		    // renderScene.clear = true;
+		    effectFXAA = new THREE.ShaderPass(THREE.FXAAShader);
+		    effectFXAA.uniforms['resolution'].value.set(1 / window.innerWidth, 1 / window.innerHeight );
```
Comment (bhouston) [inline examples/webgl_postprocessing_unreal_bloom.html]: This will have to be adjusted in the resize function.

**Comment by mrdoob** [comment]
#9351

**Pull request closed without merging**